### PR TITLE
Add resource loading support for audio files.

### PR DESCRIPTION
**Pull request by RubyNova**

### CODE REVIEW

File: src/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.cpp [hunk 1]
```diff
@@ -327,4 +330,43 @@ namespace NovelRT::ResourceManagement::Desktop
...
+
+        const size_t _bufferSize = 2048;
+        SF_INFO info;
+        info.format = 0;
+        auto filePathString = filePath.string();
+        SNDFILE* file = sf_open(filePathString.c_str(), SFM_READ, &info);
+
+        if (file == nullptr)
+        {
+            throw NovelRT::Exceptions::FileNotFoundException(filePath.string());
```
Comment (capnkenny) [inline src/NovelRT/ResourceManagement/Desktop/DesktopResourceLoader.cpp:349]: Is this entirely true here? - the original logging would log the error message returned by sndfile (whether it was malformed file, wrong/unsupported format, etc) so we should probably do the same before throwing (or we include sndfile's error in the exception we throw)

**Review by capnkenny — CHANGES_REQUESTED** [review]
LGTM bar one issue with an exception message

**Comment by RubyNova** [comment]
This PR is blocked until I can add interop support for resource loading APIs.

**Review by capnkenny — APPROVED** [review]
LGTM

**Pull request merged**